### DISCUSSION TITLE
image drawn by iterm2 backend is mis-aligned

### DISCUSSION THREAD

**Discussion by VPraharsha03** [wezterm/wezterm#2948] [Q&A] [answered]
When I specify "iTerm2" as the image backend when using neofetch, the image which is supposed to be displayed on the left is cut out when the window is in regular size and shows up in the middle of the terminal when terminal is maximized. 

```bash
neofetch --iterm2 ~/sonic.jpg --size 400px
```

![neofetch](https://user-images.githubusercontent.com/56556436/212089288-55a22d17-6350-42ad-892b-b115372a18de.png)

However it shows up properly on the left when
```bash
$ wezterm imgcat /path/to/image.png
```
is run.

Here's the debug overlay
```
Debug Overlay
wezterm version: 20221119-145034-49b9839f
OpenGL version: Intel(R) HD Graphics 2500 4.0.0 - Build 10.18.10.5100
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
19:25:30.760 WARN window::os::windows::wgl > failed to created extended OpenGL context (ChoosePixelFormatARB returned 0 formats), fall back to basic
19:36:05.843 ERROR wezterm_term::terminalstate::iterm > Unable to decode image: unknown format!?: size=0 ITermFileData { name: None, size: None, width: Pixels(400), height: Pixels(400), preserve_aspect_ratio: true, inline: true, do_not_move_cursor: false, data: [] }
19:36:13.213 ERROR wezterm_term::terminalstate::iterm > Unable to decode image: unknown format!?: size=0 ITermFileData { name: None, size: None, width: Pixels(400), height: Pixels(400), preserve_aspect_ratio: true, inline: true, do_not_move_cursor: false, data: [] }
19:38:48.537 ERROR wezterm_term::terminalstate::iterm > Unable to decode image: unknown format!?: size=0 ITermFileData { name: None, size: None, width: Pixels(400), height: Pixels(400), preserve_aspect_ratio: true, inline: true, do_not_move_cursor: false, data: [] }
```

I'm not very sure what's going on here. Any idea of what's causing this ?

I tried sixel but I don't see anything on the left

**Comment by wez** [comment, marked as answer]
Are you running in WSL?  Can you try using `wezterm ssh` to connect into your WSL instance and compare?
On Windows, the conpty layer can interfere with and/or be confused by escape sequences that it doesn't natively support.
Using `wezterm ssh` bypasses conpty and may have different results.

**Reply by VPraharsha03** [reply]
@wez Yes I'm using WSL. Yes, I tried `wezterm ssh` to ssh into the running WSL2 distro and the image was displayed properly. Is there any other workaround ?

**Reply by VPraharsha03** [reply]
The same appears to be the reason for sixel to not work, as I could even get sixel to display on SSH'ing

**Reply by wez** [reply]
re: sixel, conpty doesn't support it; here's the tracking issue in wezterm, which links to the corresponding issues for conpty:
* https://github.com/wez/wezterm/issues/1236

If you don't want to use ssh, another option is to use multiplexing, if you're using WSL 1.  WSL 2 doesn't support exposing unix domain sockets in the same way, so won't work the same way:
* https://wezfurlong.org/wezterm/multiplexing.html#connecting-into-windows-subsystem-for-linux

I'd recommend using `wezterm ssh` if you want image protocols to work their best on Windows.
